src/dune_rules/odoc.ml

```diff
@@ -791,18 +813,36 @@ let setup_lib_html_rules_def =
     f
 ;;
 
-let setup_lib_html_rules sctx lib =
-  Memo.With_implicit_output.exec setup_lib_html_rules_def (sctx, lib)
+let setup_lib_html_rules sctx ~search_db lib =
+  Memo.With_implicit_output.exec (setup_lib_html_rules_def ~search_db) (sctx, lib)
 ;;
 
 let setup_pkg_html_rules_def =
   let f (sctx, pkg) =
     let ctx = Super_context.context sctx in
+    let target = Pkg pkg in
     let* libs = libs_of_pkg ctx ~pkg in
-    let* () = Memo.parallel_iter libs ~f:(setup_lib_html_rules sctx)
+    let dir = Paths.html ctx target in
+    let* search_db =
+      let* sherlodoc_is_installed = Sherlodoc.is_installed sctx ~dir in
+      if sherlodoc_is_installed
+      then (
```

Comment:
Doing would probably result in a slight loss of performance, because the `if sherlodoc_is_installed` is duplicated but there code inside that is different. Its  certainly fine, I will give it a shot.

Reply:
No, I'm talking about the part that inside `then` as well. That's not 100% the same but there might be something to factor out (that's not about performance)

Reply:
I have made a refactoring for the installation check. For the rest I dont really see something that might be refactore.